src/app/search/search.component.ts

```diff
@@ -210,17 +201,43 @@ export class SearchComponent implements OnInit {
             this.sortModeMap.set(key, sortby);
           }
         }
-      if (this.checkboxMap.get(key).get(bucket.key)) {
-        this.entryOrder.get(key).SelectedItems.set(bucket.key, bucket.doc_count);
-      } else {
-        this.entryOrder.get(key).Items.set(bucket.key, bucket.doc_count);
+      let doc_count = bucket.doc_count;
+      if (key === 'tags.verified' && !bucket.key) {
+        doc_count = this.nonverifiedcount;
+      }
+      if (doc_count > 0) {
+        if (this.checkboxMap.get(key).get(bucket.key)) {
+          this.entryOrder.get(key).SelectedItems.set(bucket.key, doc_count);
+        } else {
+          this.entryOrder.get(key).Items.set(bucket.key, doc_count);
+        }
       }
       if (!this.setFilter) {
         this.checkboxMap.get(key).set(bucket.key, false);
       }
     });
   }
 
+  /** this function will get the actual count number of non-verified items,
+   * Not the best way to do it, try to improve this later.
```

Comment:
Could you elaborate on what is occurring here actually and how it could be improved?